test/integration/reader-order.spec.ts

```diff
@@ -0,0 +1,184 @@
+             * 2. If ean_reader returns null, upc_reader.decode() is called
+             * 3. If upc_reader returns null, code_128_reader.decode() is called
+             * 4. First reader to return a non-null result wins
+             * 
+             * This is implemented in barcode_decoder.js tryDecode():
+             *   for (i = 0; i < _barcodeReaders.length && result === null; i++) {
+             *       result = _barcodeReaders[i].decodePattern(barcodeLine.line);
+             *   }
+             */
+            expect(true).to.be.true;
+        });
+    });
+    
+    describe('External reader order documentation', () => {
+        it('should document that external readers follow the same ordering rules', () => {
+            /**
+             * External readers are added to the READERS registry via registerReader().
+             * Once registered, they can be used in the readers config array.
+             * 
+             * The key point is:
+             * - External readers must be REGISTERED before they can be listed in config.readers
+             * - Their position in config.readers determines their priority, just like internal readers
+             * - There is no automatic "internal first, external second" ordering
+             * 
+             * Example:
+             *   Quagga.registerReader('my_custom_reader', MyCustomReader);
+             *   
+             *   Quagga.init({
+             *     decoder: {
+             *       readers: ['my_custom_reader', 'ean_reader', 'code_128_reader']
+             *     }
+             *   });
+             * 
+             * In this case, my_custom_reader is tried FIRST because it's first in the array.
+             */
+            expect(true).to.be.true;
+        });
+    });
+});
+
+describe('Priority Behavior with Multiple Readers', () => {
+    it('should demonstrate that reader order affects which barcode format is detected', async function() {
+        this.timeout(20000);
+        
+        // This test uses a real EAN-13 barcode image and verifies it decodes correctly
+        // when ean_reader is prioritized over upc_e_reader
+        const config = {
+            inputStream: {
+                size: 640,
+            },
+            locator: {
+                patchSize: 'medium',
+                halfSample: true,
+            },
+            numOfWorkers: 0,
+            decoder: {
+                // EAN reader first - should decode EAN-13 barcodes as ean_13
+                readers: ['ean_reader', 'upc_e_reader', 'upc_reader'],
+            },
+            locate: true,
+            src: `${typeof window !== 'undefined' ? '/' : ''}test/fixtures/ean/image-001.jpg`,
+        };
+        
+        const result = await Quagga.decodeSingle(config);
+        
+        expect(result).to.be.an('object');
+        expect(result.codeResult).to.be.an('object');
+        expect(result.codeResult.format).to.equal('ean_13');
```

Comment:
i'm not sure that testing a EAN barcode with EAN first demonstrates the priority working correctly -- wouldn't it make more sense to use a barcode, like ean-5 or ean-8 that the ean reader can read in multiple configurations, and use that to demonstrate priority?